src/neural/cuda/layers.cc

```diff
@@ -1677,7 +1691,7 @@ void AttentionPolicyHead<DataType>::Eval(
   DataType* promotion_logits = output + 64 * 64;
 
   ComputePromotionLogits<DataType>(N, policy_d_model_, promotion_logits,
-                                   scratch2, ip4_pol_w_, output, stream);
+                                   scratch0, ip4_pol_w_, output, stream);
```

Comment:
There was a bug report that promotions are broken with this PR.  Looking at the code I think that scratch0 here should actually be wk.

Reply:
Kovax tried out this fix and it seemed to work.

Reply:
sorry about that! Yeah, I can confirm that was unintentional.